File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-doc-5-4/QueryParameter.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class QueryParameter : ISerialized
    {
        public QueryParameter(XmlNode xmlNode)
        {
            XmlNode queryParameterIdNode = xmlNode.SelectSingleNode("queryParameterId");
            
            if (queryParameterIdNode != null)
            {
                if (queryParameterIdNode.Attributes["href"] != null || queryParameterIdNode.Attributes["id"] != null) 
                {
                    if (queryParameterIdNode.Attributes["id"] != null) 
                    {
                        queryParameterIdIDRef_ = queryParameterIdNode.Attributes["id"].Value;
                        QueryParameterId ob = new QueryParameterId(queryParameterIdNode);
                        IDManager.SetID(queryParameterIdIDRef_, ob);
                    }
                    else if (queryParameterIdNode.Attributes["href"] != null)
                    {
                        queryParameterIdIDRef_ = queryParameterIdNode.Attributes["href"].Value;
                    }
                    else
                    {
                        queryParameterId_ = new QueryParameterId(queryParameterIdNode);
                    }
                }
                else
                {
                    queryParameterId_ = new QueryParameterId(queryParameterIdNode);
                }
            }
            
        
            XmlNode queryParameterValueNode = xmlNode.SelectSingleNode("queryParameterValue");
            
            if (queryParameterValueNode != null)
            {
                if (queryParameterValueNode.Attributes["href"] != null || queryParameterValueNode.Attributes["id"] != null) 
                {
                    if (queryParameterValueNode.Attributes["id"] != null) 
                    {
                        queryParameterValueIDRef_ = queryParameterValueNode.Attributes["id"].Value;
                        XsdTypeNormalizedString ob = new XsdTypeNormalizedString(queryParameterValueNode);
                        IDManager.SetID(queryParameterValueIDRef_, ob);
                    }
                    else if (queryParameterValueNode.Attributes["href"] != null)
                    {
                        queryParameterValueIDRef_ = queryParameterValueNode.Attributes["href"].Value;
                    }
                    else
                    {
                        queryParameterValue_ = new XsdTypeNormalizedString(queryParameterValueNode);
                    }
                }
                else
                {
                    queryParameterValue_ = new XsdTypeNormalizedString(queryParameterValueNode);
                }
            }
            
        
            XmlNode queryParameterOperatorNode = xmlNode.SelectSingleNode("queryParameterOperator");
            
            if (queryParameterOperatorNode != null)
            {
                if (queryParameterOperatorNode.Attributes["href"] != null || queryParameterOperatorNode.Attributes["id"] != null) 
                {
                    if (queryParameterOperatorNode.Attributes["id"] != null) 
                    {
                        queryParameterOperatorIDRef_ = queryParameterOperatorNode.Attributes["id"].Value;
                        QueryParameterOperator ob = new QueryParameterOperator(queryParameterOperatorNode);
                        IDManager.SetID(queryParameterOperatorIDRef_, ob);
                    }
                    else if (queryParameterOperatorNode.Attributes["href"] != null)
                    {
                        queryParameterOperatorIDRef_ = queryParameterOperatorNode.Attributes["href"].Value;
                    }
                    else
                    {
                        queryParameterOperator_ = new QueryParameterOperator(queryParameterOperatorNode);
                    }
                }
                else
                {
                    queryParameterOperator_ = new QueryParameterOperator(queryParameterOperatorNode);
                }
            }
            
        
        }
        
    
        #region QueryParameterId_
        private QueryParameterId queryParameterId_;
        public QueryParameterId QueryParameterId_
        {
            get
            {
                if (this.queryParameterId_ != null)
                {
                    return this.queryParameterId_; 
                }
                else if (this.queryParameterIdIDRef_ != null)
                {
                    queryParameterId_ = IDManager.getID(queryParameterIdIDRef_) as QueryParameterId;
                    return this.queryParameterId_; 
                }
                else
                {
                      return this.queryParameterId_; 
                }
            }
            set
            {
                if (this.queryParameterId_ != value)
                {
                    this.queryParameterId_ = value;
                }
            }
        }
        #endregion
        
        public string queryParameterIdIDRef_ { get; set; }
        #region QueryParameterValue_
        private XsdTypeNormalizedString queryParameterValue_;
        public XsdTypeNormalizedString QueryParameterValue_
        {
            get
            {
                if (this.queryParameterValue_ != null)
                {
                    return this.queryParameterValue_; 
                }
                else if (this.queryParameterValueIDRef_ != null)
                {
                    queryParameterValue_ = IDManager.getID(queryParameterValueIDRef_) as XsdTypeNormalizedString;
                    return this.queryParameterValue_; 
                }
                else
                {
                      return this.queryParameterValue_; 
                }
            }
            set
            {
                if (this.queryParameterValue_ != value)
                {
                    this.queryParameterValue_ = value;
                }
            }
        }
        #endregion
        
        public string queryParameterValueIDRef_ { get; set; }
        #region QueryParameterOperator_
        private QueryParameterOperator queryParameterOperator_;
        public QueryParameterOperator QueryParameterOperator_
        {
            get
            {
                if (this.queryParameterOperator_ != null)
                {
                    return this.queryParameterOperator_; 
                }
                else if (this.queryParameterOperatorIDRef_ != null)
                {
                    queryParameterOperator_ = IDManager.getID(queryParameterOperatorIDRef_) as QueryParameterOperator;
                    return this.queryParameterOperator_; 
                }
                else
                {
                      return this.queryParameterOperator_; 
                }
            }
            set
            {
                if (this.queryParameterOperator_ != value)
                {
                    this.queryParameterOperator_ = value;
                }
            }
        }
        #endregion
        
        public string queryParameterOperatorIDRef_ { get; set; }
        
    
        
    
    }
    
}

File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-doc-5-4/QueryParameter.cpp
// QueryParameter.cpp 
#include "QueryParameter.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

QueryParameter::QueryParameter(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //queryParameterIdNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* queryParameterIdNode = xmlNode->FirstChildElement("queryParameterId");

   if(queryParameterIdNode){queryParameterIdIsNull_ = false;}
   else{queryParameterIdIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- queryParameterIdNode , address : " << queryParameterIdNode << std::endl;
   #endif
   if(queryParameterIdNode)
   {
      if(queryParameterIdNode->Attribute("href") || queryParameterIdNode->Attribute("id"))
      {
          if(queryParameterIdNode->Attribute("id"))
          {
             queryParameterIdIDRef_ = queryParameterIdNode->Attribute("id");
             queryParameterId_ = boost::shared_ptr<QueryParameterId>(new QueryParameterId(queryParameterIdNode));
             queryParameterId_->setID(queryParameterIdIDRef_);
             IDManager::instance().setID(queryParameterIdIDRef_,queryParameterId_);
          }
          else if(queryParameterIdNode->Attribute("href")) { queryParameterIdIDRef_ = queryParameterIdNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { queryParameterId_ = boost::shared_ptr<QueryParameterId>(new QueryParameterId(queryParameterIdNode));}
   }

   //queryParameterValueNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* queryParameterValueNode = xmlNode->FirstChildElement("queryParameterValue");

   if(queryParameterValueNode){queryParameterValueIsNull_ = false;}
   else{queryParameterValueIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- queryParameterValueNode , address : " << queryParameterValueNode << std::endl;
   #endif
   if(queryParameterValueNode)
   {
      if(queryParameterValueNode->Attribute("href") || queryParameterValueNode->Attribute("id"))
      {
          if(queryParameterValueNode->Attribute("id"))
          {
             queryParameterValueIDRef_ = queryParameterValueNode->Attribute("id");
             queryParameterValue_ = boost::shared_ptr<XsdTypeNormalizedString>(new XsdTypeNormalizedString(queryParameterValueNode));
             queryParameterValue_->setID(queryParameterValueIDRef_);
             IDManager::instance().setID(queryParameterValueIDRef_,queryParameterValue_);
          }
          else if(queryParameterValueNode->Attribute("href")) { queryParameterValueIDRef_ = queryParameterValueNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { queryParameterValue_ = boost::shared_ptr<XsdTypeNormalizedString>(new XsdTypeNormalizedString(queryParameterValueNode));}
   }

   //queryParameterOperatorNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* queryParameterOperatorNode = xmlNode->FirstChildElement("queryParameterOperator");

   if(queryParameterOperatorNode){queryParameterOperatorIsNull_ = false;}
   else{queryParameterOperatorIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- queryParameterOperatorNode , address : " << queryParameterOperatorNode << std::endl;
   #endif
   if(queryParameterOperatorNode)
   {
      if(queryParameterOperatorNode->Attribute("href") || queryParameterOperatorNode->Attribute("id"))
      {
          if(queryParameterOperatorNode->Attribute("id"))
          {
             queryParameterOperatorIDRef_ = queryParameterOperatorNode->Attribute("id");
             queryParameterOperator_ = boost::shared_ptr<QueryParameterOperator>(new QueryParameterOperator(queryParameterOperatorNode));
             queryParameterOperator_->setID(queryParameterOperatorIDRef_);
             IDManager::instance().setID(queryParameterOperatorIDRef_,queryParameterOperator_);
          }
          else if(queryParameterOperatorNode->Attribute("href")) { queryParameterOperatorIDRef_ = queryParameterOperatorNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { queryParameterOperator_ = boost::shared_ptr<QueryParameterOperator>(new QueryParameterOperator(queryParameterOperatorNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<QueryParameterId> QueryParameter::getQueryParameterId()
{
   if(!this->queryParameterIdIsNull_){
        if(queryParameterIdIDRef_ != ""){
             return boost::shared_static_cast<QueryParameterId>(IDManager::instance().getID(queryParameterIdIDRef_));
        }else{
             return this->queryParameterId_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<QueryParameterId>();
   }
}
boost::shared_ptr<XsdTypeNormalizedString> QueryParameter::getQueryParameterValue()
{
   if(!this->queryParameterValueIsNull_){
        if(queryParameterValueIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeNormalizedString>(IDManager::instance().getID(queryParameterValueIDRef_));
        }else{
             return this->queryParameterValue_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeNormalizedString>();
   }
}
boost::shared_ptr<QueryParameterOperator> QueryParameter::getQueryParameterOperator()
{
   if(!this->queryParameterOperatorIsNull_){
        if(queryParameterOperatorIDRef_ != ""){
             return boost::shared_static_cast<QueryParameterOperator>(IDManager::instance().getID(queryParameterOperatorIDRef_));
        }else{
             return this->queryParameterOperator_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<QueryParameterOperator>();
   }
}
}

